src-tauri/src/commands/config.rs

```diff
@@ -58,6 +60,21 @@ pub async fn set_install_directory(
   })
 }
 
+#[tauri::command]
+pub async fn set_flatpak_install_directory(
+  config: tauri::State<'_, tokio::sync::Mutex<LauncherConfig>>,
+) -> Result<Option<String>, CommandError> {
+  let xdg_home_data = env::var("XDG_DATA_HOME").unwrap();
```

Comment:
this could be:

let new_dir = env::var("XDG_DATA_HOME")
    .map(PathBuf::from)
    .map(|path| path.to_string_lossy().into_owned())
    .unwrap();